components/feature/session/src/main/java/mozilla/components/feature/session/SessionUseCases.kt

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.session
+
+import android.net.Uri
+import android.text.TextUtils
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.Engine
+
+/**
+ * Contains all use cases related to the session feature. Every public function represents a single
+ * use case.
+ */
+class SessionUseCases(
```

Comment:
Initially I was thinking about one usecase == one class. Couldn't this class get potentially quite big and complex?

Reply:
However with the typealias approach below you could try splitting the use cases into multiple classes once they get too big.

Reply:
Yes, definitely didn't mean to imply everything has to be in one class. Originally had nested classes, but could also be a set of top-level classes.